packages/trees/frontend/sirius-components-trees/src/treeitems/TreeItemDirectEditInput.tsx

```diff
@@ -0,0 +1,164 @@
+            }
+          }
+        }
+      }
+    }
+  }
+`;
+
+export const TreeItemDirectEditInput = ({
+  editingContextId,
+  treeId,
+  treeItemId,
+  refParent,
+  editingkey,
+  onClose,
+}: TreeItemDirectEditInputProps) => {
+  const [state, setState] = useState({
+    newLabel: '',
+    editingkey: editingkey,
+  });
+
+  const {
+    loading: initialLabelTreeItemLoading,
+    data: initialLabelTreeItemItemData,
+    error: initialLabelTreeItemItemError,
+  } = useQuery<GQLInitialDirectEditElementLabelData, GQLInitialDirectEditElementLabelInput>(
+    initialDirectEditElementLabeQuery,
+    {
+      variables: {
+        editingContextId: editingContextId,
+        representationId: treeId,
+        treeItemId: treeItemId,
+      },
+    }
+  );
+
+  useEffect(() => {
+    if (
+      !initialLabelTreeItemLoading &&
+      !initialLabelTreeItemItemError &&
+      initialLabelTreeItemItemData?.viewer.editingContext.representation.description.initialDirectEditTreeItemLabel
+    ) {
+      const initialLabel =
+        initialLabelTreeItemItemData?.viewer.editingContext.representation.description.initialDirectEditTreeItemLabel;
+      setState((prevState) => {
+        return { ...prevState, newLabel: initialLabel + state.editingkey };
+      });
+    }
+  }, [initialLabelTreeItemLoading, initialLabelTreeItemItemError, initialLabelTreeItemItemData]);
+
+  const [renameTreeItem, { loading: renameTreeItemLoading, data: renameTreeItemData, error: renameTreeItemError }] =
+    useMutation(renameTreeItemMutation);
+
+  useEffect(() => {
+    if (!renameTreeItemLoading && !renameTreeItemError && renameTreeItemData?.renameTreeItem) {
+      const { renameTreeItem } = renameTreeItemData;
+      if (renameTreeItem.__typename === 'SuccessPayload') {
+        onClose();
+      }
+    }
+  }, [renameTreeItemData, renameTreeItemError, renameTreeItemLoading]);
+
+  const doRename = () => {
+    const isNameValid = state.newLabel.length >= 1;
+    if (isNameValid) {
+      renameTreeItem({
+        variables: {
+          input: {
+            id: crypto.randomUUID(),
+            editingContextId: editingContextId,
+            representationId: treeId,
+            treeItemId: treeItemId,
+            newLabel: state.newLabel,
+          },
+        },
+      });
+    } else {
+      onClose();
+    }
+  };
+
+  const handleChange = (event) => {
+    const newLabel = event.target.value;
+    setState((prevState) => {
+      return { ...prevState, newLabel: newLabel };
+    });
+  };
+
+  const onFinishEditing = (event) => {
+    const { key } = event;
+    if (key === 'Enter') {
+      doRename();
+    } else if (key === 'Escape') {
+      onClose();
+    }
+  };
+
+  const onFocusIn = (event) => event.target.select();
+
+  const onFocusOut = (event) => {
+    //Prevents calling doRename when pressing Escape
+    if (!event.relatedTarget || event.relatedTarget != refParent) {
+      doRename();
+    }
```

Comment:
- Why not detect `Esc`?
- Do you really need the reference of the parent here?

Reply:
This event is not fired when the user press Esc but when the component unmount after onClose() and onBlur is called.